apps/studio/src/features/editing-experience/components/FormBuilder.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  FormControl,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInputField,
+  NumberInputStepper,
+  RadioGroup,
+  Select,
+} from '@chakra-ui/react'
+import {
+  Button,
+  FormLabel,
+  Input,
+  NumberInput,
+  Radio,
+  Switch,
+} from '@opengovsg/design-system-react'
+import { IsomerComponent } from '@opengovsg/isomer-components'
+import { useForm } from 'react-hook-form'
+import IsomerSchema from './0.1.0.json'
+
+// TODO: Props here might not make sense, need to check again
+export interface FormBuilderProps {
+  component: keyof typeof IsomerSchema.components.complex
+  data?: IsomerComponent
+}
+
+// Strategy:
+// - Read the sub-schema for the given component
+// - For each prop inside the sub-schema, render the appropriate input field
+//
+// TODO:
+// - Support type = 'array', format = 'prose' to be Tiptap editor
+// - Support type = 'array' with items of type = 'object' to be sub-form (no drag and drop yet)
+// - Support $ref to other parts of the schema
+// - Handle grouping of fields
+export const FormBuilder = ({
+  component,
+  data,
+}: FormBuilderProps): JSX.Element => {
+  const schema = IsomerSchema.components.complex[component]
+  const { required, properties } = schema
+  const { register, handleSubmit } = useForm<typeof properties>()
+
+  // TODO: Remove or replace with proper submit function
+  const onSubmit = handleSubmit((data) => console.log(data))
+
+  return (
+    <div className="bg-white p-4">
+      <form onSubmit={onSubmit}>
+        {Object.keys(properties)
+          .filter((prop) => prop !== 'type')
+          .map((prop) => {
+            const {
+              type,
+              format,
+              enum: options,
+              title,
+              description,
+              // @ts-ignore
+            } = properties[prop]
+
+            if (type === 'string' && options && format === 'select') {
+              return (
+                <div className="py-2">
```

Comment:
prefer use of `Box` since we use fully chakra components

Reply:
Ah good point, updated in 7429b65c56669cc4c10c9e532600212c66fee135.